src/main/java/com/microsoft/azure/functions/worker/handler/FunctionWarmupHandler.java

```diff
@@ -0,0 +1,82 @@
+package com.microsoft.azure.functions.worker.handler;
+
+import com.microsoft.azure.functions.rpc.messages.*;
+import com.microsoft.azure.functions.worker.WorkerLogManager;
+import com.microsoft.azure.functions.worker.broker.JavaFunctionBroker;
+import com.microsoft.azure.functions.worker.reflect.FactoryClassLoader;
+
+import java.util.*;
+import java.util.logging.Level;
+
+
+public class FunctionWarmupHandler extends MessageHandler<FunctionWarmupRequest, FunctionWarmupResponse.Builder> {
+
+    private static final String WARM_UP_FUNCTION_NAME = "WarmupFunc";
+    private static final String WARM_UP_FUNCTION_ENTRY_POINT = "com.microsoft.azure.functions.warmup.java.Function.run";
+    private static final String WARM_UP_FUNCTION_SCRIPT_FILE = "/annotationLib/warmup-httptrigger.jar";
+    private final JavaFunctionBroker javaFunctionBroker = new JavaFunctionBroker(new FactoryClassLoader().createClassLoaderProvider());
+
+    public FunctionWarmupHandler() {
+        super(StreamingMessage::getFunctionWarmupRequest,
+                FunctionWarmupResponse::newBuilder,
+                FunctionWarmupResponse.Builder::setResult,
+                StreamingMessage.Builder::setFunctionWarmupResponse);
+    }
+
+    @Override
+    String execute(FunctionWarmupRequest functionWarmupRequest, FunctionWarmupResponse.Builder builder) {
+        try {
+            WorkerLogManager.getSystemLogger().info("azure function java worker warm up start.");
+            this.javaFunctionBroker.setWorkerDirectory(functionWarmupRequest.getWorkerDirectory());
+            warmupFunctionEnvironmentReloadRequestHandler(functionWarmupRequest);
+            UUID functionId = warmupFunctionLoadRequestHandler(functionWarmupRequest);
+            warmupInvocationRequestHandler(functionId);
+            WorkerLogManager.getSystemLogger().info("azure function java worker warm up completed successfully.");
+        } catch (Exception e) {
+            WorkerLogManager.getSystemLogger().severe("warm up process failed with exception: " + e.getMessage());
+            throw new RuntimeException(e);
+        }
+        return "azure function java worker warm up completed";
+    }
+
+    private void warmupFunctionEnvironmentReloadRequestHandler(FunctionWarmupRequest functionWarmupRequest) throws Exception {

```

Comment:
looks like arg `functionWarmupRequest` unused

```suggestion
    private void runFunctionEnvironmentReload(FunctionWarmupRequest functionWarmupRequest) throws Exception {

```